julearn/model_selection/tests/test_skopt_searcher.py

```diff
@@ -0,0 +1,134 @@
+"""Provides tests for the bayes searcher."""
+
+# Authors: Federico Raimondo <f.raimondo@fz-juelich.de>
+# License: AGPL
+from typing import Dict
+
+import pytest
+
+from julearn.model_selection._skopt_searcher import (
+    _prepare_skopt_hyperparameters_distributions,
+)
+
+
+sksp = pytest.importorskip("skopt.space")
+
+@pytest.mark.parametrize(
+    "params_to_tune,expected_types, expected_dist",
+    [
+        (
+            {
+                "n_components": (0.2, 0.7, "uniform"),
+                "n_neighbors": (1.0, 10.0, "log-uniform"),
+            },
+            ("float", "float"),
+            ("uniform", "log-uniform"),
+        ),
+        (
+            {
+                "n_components": (1, 20, "uniform"),
+                "n_neighbors": (1, 10, "log-uniform"),
+            },
+            ("int", "int", "int"),
+            ("uniform", "log-uniform"),
+        ),
+        (
+            {
+                "options": (True, False, "categorical"),
+                "more_options": ("a", "b", "c", "d", "categorical"),
+            },
+            (None, None),
+            ("categorical", "categorical"),
+        ),
+        (
+            {
+                "n_components": sksp.Real(0.2, 0.7, prior="uniform"),
+                "n_neighbors": sksp.Real(1.0, 10.0, prior="log-uniform"),
+            },
+            ("float", "float"),
+            ("uniform", "log-uniform"),
+        ),
+        (
+            {
+                "n_components": sksp.Integer(1, 20, prior="uniform"),
+                "n_neighbors": sksp.Integer(1, 10, prior="log-uniform"),
+            },
+            ("int", "int"),
+            ("uniform", "log-uniform"),
+        ),
+        (
+            {
+                "options": sksp.Categorical([True, False]),
+                "more_options": sksp.Categorical(
+                    ("a", "b", "c", "d"),
+                ),
+            },
+            (None, None),
+            ("categorical", "categorical"),
+        ),
+    ],
+)
+def test__prepare_skopt_hyperparameters_distributions(
```

Comment:
The typing annotations need to be adjusted similar to my comments in test_optuna_searcher.py